src/Databases/DatabaseOnDisk.cpp

```diff
@@ -352,6 +353,14 @@ void DatabaseOnDisk::dropTable(ContextPtr local_context, const String & table_na
     fs::remove(table_metadata_path_drop);
 }
 
+UUID DatabaseOnDisk::GetTableUUIDFromDetachedMetadata(ContextPtr local_context, const String & table_metadata_path) const
+{
+    ASTPtr ast_detached = parseQueryFromMetadata(log, local_context, table_metadata_path);
+    auto & create_detached = ast_detached->as<ASTCreateQuery &>();
+
+    return create_detached.uuid;
```

Comment:
Have we checked that all table engines we can use with DatabaseAtomic have a uuid?

Reply:
Althought Im not sure I think all tables must have a uuid. Pls let me know if its not. 
Catalog doesn't support removing tables without uuid.

Reply:
For example: `Log` engines have a zero `uuid`, but such tables can be used only with `Lazy` databases.

```
SELECT
    uuid,
    engine_full
FROM system.tables
WHERE database = 'lz'

Query id: 90b23720-d899-49ff-a888-01025fff6c60

Row 1:
──────
uuid:        00000000-0000-0000-0000-000000000000
engine_full: StripeLog SETTINGS storage_policy = 'object_storage'
```

I mean I don't know for sure if every table engine which can be used with atomic database have an uuid.  
Anyway suppose that it will be great to cover as much as possible table engines in the test, not only `*MergeTree` ones.

Reply:
I added some engines. If need add anything else - let me know.